test:conditional-evaluation-api-tests

GeethaParthasarathy opened the pull request:
## Description

Adds comprehensive API test coverage for the POST /v2/conditionals/evaluation endpoint.

Success cases (6 tests):

- Basic conditional evaluation with matching condition
- Multiple conditions matching
- No matching conditions (empty result)
- Evaluation with tenant ID
- Evaluation by process definition key
- Response schema validation

Error cases (6 tests):

- Unauthorized request (401)
- Bad request with missing variables (400)
- Invalid tenant (400)
- Invalid process definition key (404)
- Unsupported media type (415)
- Invalid JSON (400)

## Checklist

<!--- Please delete options that are not relevant. Boxes should be checked by reviewer. -->
- [ ] Enable backports when necessary (fex. [for bug fixes](https://github.com/camunda/camunda/blob/main/CONTRIBUTING.md#backporting-changes) or [for CI changes](https://github.com/camunda/camunda/wiki/CI-&-Automation#when-to-backport-ci-changes)).

## Related issues

https://github.com/camunda/product-hub/issues/3281

closes #

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/camunda/camunda?pullRequest=43270) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/camunda/camunda?pullRequest=43270) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/camunda/camunda?pullRequest=43270) it.</sub>

GeethaParthasarathy commented:
Tagging CI workflow for reference : https://github.com/camunda/camunda/actions/runs/20488891204

berkaycanbc commented:
FYI, @GeethaParthasarathy I will review this once I am back from holidays.

berkaycanbc commented on qa/c8-orchestration-cluster-e2e-test-suite/tests/api/v2/conditional/conditional-evaluation-api-tests.spec.ts:
```diff
@@ -0,0 +1,326 @@
+    const json = await res.json();
+    assertRequiredFields(json, conditionalEvaluationResponseRequiredFields);
+    expect(json.processInstances).toBeDefined();
+
```
🔧 Same as above; we can still assert `conditionalProcessInstanceItemRequiredFields` for both started instances.

berkaycanbc reviewed (changes requested):
Thanks for the PR @GeethaParthasarathy! 🚀 

Please note that I've reviewed the PR according to the [Review Emoji Code](https://github.com/camunda/camunda/blob/main/CONTRIBUTING.md#review-emoji-code). Please refer to the page to get insights of the emojis.

I've requested some changes in comments and below. Please take a look. 👀 

🔧 I've seen test cases only for full match or non match. It would be good to add a test case where condition is partially matched and verify only matched definition's instance is started.

🔧 Required fields in the request body are not tested. Would be proper to add test cases verifying the request is rejected when a required field in the request is missing.


FYI @hilalbursalii I only reviewed the tests themselves but not any convention such as naming, package locations etc.

GeethaParthasarathy committed:
test: add conditional evaluation API tests with fixes

- Add conditional start event BPMN resources
- Add conditional evaluation API v2 tests
- Fix 409 conflict by using conditional evaluation API to get process definition key
- Fix authorization assertions to use group name instead of groupId
- Assert required fields for all process instances
- Add nodejs version configuration for asdf

GeethaParthasarathy committed:
revert: remove identity test changes

Main branch already includes the correct identity test fixes.
Keeping only conditional evaluation API test additions.

hilalbursalii commented on qa/c8-orchestration-cluster-e2e-test-suite/tests/api/v2/conditional/conditional-evaluation-api-tests.spec.ts:
```diff
@@ -0,0 +1,363 @@
+    expect(json.title).toBeDefined();
+    expect(json.status).toBe(400);
+    expect(json.detail).toBeDefined();
+    expect(json.instance).toBeDefined();
```
Let's use `assertBadRequest` function in `/utils/http.ts` and make adjustments in the function if necessary.

hilalbursalii commented on qa/c8-orchestration-cluster-e2e-test-suite/utils/beans/requestBeans.ts:
```diff
@@ -635,3 +635,66 @@ export function CREATE_CLUSTER_VARIABLE() {
+  processDefinitionKey: string,
+) {
+  return {
+    processDefinitionKey,
```
Forgot to mention that in our call. Beans file can be longer so could you create a new bean file just like `element-instance-requestBeans.ts` under beans and move those values to there?

hilalbursalii commented on qa/c8-orchestration-cluster-e2e-test-suite/tests/api/v2/conditional/conditional-evaluation-api-tests.spec.ts:
```diff
@@ -0,0 +1,363 @@
+
+      for (const instance of json.processInstances) {
+        await cancelProcessInstance(instance.processInstanceKey);
+      }
```
Could we move the cleanup in `afterAll` section? If the test fails, we can't clean them up. The suggestion applies to other cases as well.

hilalbursalii reviewed (changes requested):
Great work @GeethaParthasarathy 👏 
Overall, it's pretty nice 😍 ! I left a couple of comments to improve it. Please let me know if anything is unclear.

GeethaParthasarathy committed:
refactor: improve conditional evaluation test cleanup and organization

Move conditional evaluation request beans to dedicated file for better organization. Add afterAll hook to cleanup process instances instead of inline cleanup calls. Remove unnecessary if statement from schema verification test and expect array length to be greater than 0 for process instances.

hilalbursalii commented:
> @hilalbursalii Thanks a lot for the review !! I have attended to your review comments could you pls have at look at the PR again !!

Thanks for the quick changes! All good, could you trigger the on-demand run after fixes and include the run link to PR?
 Also, could you add this new test case to [Confluence page](Docker Compose with Camunda Standalone (Zeebe, Tasklist, Operate, Identity)) below?
<img width="687" height="134" alt="Screenshot 2026-01-07 at 08 32 27" src="https://github.com/user-attachments/assets/818a26c1-02c7-4abb-8507-1e72373dd86f" />

GeethaParthasarathy commented:
> Thanks for the quick changes! All good, could you trigger the on-demand run after fixes and include the run link to PR?
> Also, could you add this new test case to [Confluence page](Docker Compose with Camunda Standalone (Zeebe, Tasklist, Operate, Identity)) below?

Hilal here is the latest workflow run link :https://github.com/camunda/camunda/actions/runs/20744771063

The link to confulence page is not actually working could you pls re-paste it and also approve the PR from your end pls . BTw thanks for all the support and detailed review for this PR 🎖️ 

@berkaycanbc If you are also done with reviewing the PR with my changes could you pls approve the PR from your end as well. TIA :)

hilalbursalii commented:
here is the link: https://confluence.camunda.com/spaces/HAN/pages/262669277/Automated+Testing+Distributions+Overview#AutomatedTestingDistributionsOverview-TestCoverageSummary%3A

berkaycanbc reviewed (approved):
Thanks for the updates @GeethaParthasarathy, LGTM! 🚀
